sbt-plugin/src/main/scala/org/scalajs/sbtplugin/ScalaJSPlugin.scala

```diff
@@ -28,9 +33,45 @@ import org.scalajs.jsenv.jsdomnodejs.JSDOMNodeJSEnv
 import org.scalajs.jsenv.phantomjs.PhantomJSEnv
 
 object ScalaJSPlugin extends AutoPlugin {
-  /** The global Scala.js IR cache */
+  /** The global Scala.js IR cache.
+   *
+   *  If you want to allocate a new cache for your purposes, prefer calling
+   *  [[newIRCache]] rather than
+   *  [[org.scalajs.core.tools.io.IRFileCache.newCache globalIRCache.newCache]].
+   *  The former will automatically register the new cache to be freed when the
+   *  build is unloaded. If you use the latter, it is your responsibility to
+   *  free the cache.
+   */
   val globalIRCache: IRFileCache = new IRFileCache()
 
+  private val allocatedIRCaches =
+    new AtomicReference[List[globalIRCache.Cache]](Nil)
+
+  /** Allocates a new IR cache linked to the [[globalIRCache]].
+   *
+   *  The allocated IR cache will automatically be freed when the build is
+   *  unloaded.
+   */
+  def newIRCache: globalIRCache.Cache = {
```

Comment:
When should one use `scalaJSIRCache`, when this method?

Reply:
Unless one knows what they're doing, they probably don't want to use `scalaJSIRCache`. Either they want to read `scalaJSIR.value` which already uses `scalaJSIRCache.value`, or they want their own `Cache`. It is unadvisable to use the same `Cache` for two different purposes, since the calls to `cached(files1)` and `cached(files2)` will interfere with each other.

So unless you want the precise set of files returned by `scalaJSIR.value`, you probably want to use `newIRCache`.

Reply:
Maybe we should just hide this for now?

Reply:
OK